core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -175,6 +176,7 @@ class CivilizationInfo {
     fun isCityState(): Boolean = nation.isCityState()
     val cityStateType: CityStateType get() = nation.cityStateType!!
     var cityStatePersonality: CityStatePersonality = CityStatePersonality.Neutral
+    var cityStateResource: TileResource? = null
```

Comment:
Since this is referencing a resource in the ruleset, the reference should be by name, as a string.